Release 2.1.2-rc1

osalyk opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/daos-stack/pmdk/18)
<!-- Reviewable:end -->

jolivier23 commented on pmdk.spec:
```diff
@@ -738,6 +738,9 @@ cp utils/pmdk.magic %{buildroot}%{_datadir}/pmdk/
+* Tue Jul 06 2021 Jeff Olivier <jeffrey.v.olivier@intel.com> - 1.11.0-2
+- Update to official release 1.11.0
+
 * Fri Jun 11 2021 Jeff Olivier <jeffrey.v.olivier@intel.com> - 1.11.0-1
```
I messed this one up...it should have had an rc1 in it.  That is why I went to release 2

osalyk committed:
test: disable VARIANT_MOVDIR64B under Valgrind

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

grom72 committed:
common: pmem2_badblock_next is replaced by pmem2_badblock_next_internal

pmem2_badblock_next is no longer used as libpmem uses
pmem2_badblock_next_internal.

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@intel.com>
Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

grom72 committed:
common: fix clang '-Wcast-function-type-strict' issue

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>
Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

grom72 committed:
common: fix - clang-format update from 9.0 to 14.0

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>
Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

grom72 committed:
common: regular validation with basic OSes

Regular run main.yml workflow to avoid incidental PR
validation failures as described in
https://github.com/pmem/pmdk/issues/6134

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>
Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

grom72 committed:
common: remove paralel execution to stop build on the first failure

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>
Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

grom72 committed:
common: add status badge for Main workflow

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>
Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

grom72 committed:
common: ChangeLog supplement for #6117 and #6127

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>
Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

grom72 committed:
fix: review

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>
Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: change lower_limit

Two functions resized:
pmemobj_create() 288->304
heap_populate_bucket() 112->128

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: check clang version

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

grom72 committed:
Include HPE in general license files

Signed-off-by: Tomasz Gromadzki <tomasz.gromadzki@hpe.com>
Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: 2.1.1-rc1 release

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: git versions

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: 2.1.1-rc2 release

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: git versions

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: 2.1.1 release

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: git versions

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
test: call setup before require_sds

It must be called after setup because it relies
on the contents of PMEMOBJ_CONF which is set there.

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
test: unblock obj_sds tests for non-pmem

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
test: check if mock functions are used

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

janekmi committed:
common: update before install (aarch64)

Signed-off-by: Jan Michalski <jan-marian.michalski@hpe.com>
Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: update workflow files

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
test: add unit test for sds_check (#1)

+ minor adjustments to error/warning messages

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

janekmi committed:
common: expand the sds.at_create CTL to also cover pmemobj_open()

https://daosio.atlassian.net/browse/DAOS-17449

Ref: daos-stack/pmdk#5

Unclean cherry-pick of 0bf7f8a565aae101f17aec4bf1c168bda4fa1243

Signed-off-by: Jan Michalski <jan-marian.michalski@hpe.com>
Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
utils: switch self hosted runners to the new repo location
replace the outdated include command with the import_tasks command

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: fix typo

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: split the pmem_tests.yml file (#9)

Due to limited hardware availability, the previous monolithic workflow could not meet the 24-hour limit imposed by GitHub.

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: reshuffle the jobs to fit both workflows into the 24-hour window

Reshuffle the jobs to fit a workflow containing the longest-running jobs into a 24-hour window.

Hence there is no longer a clear cut (with Valgrind vs without Valgrind).
They are simply called part 1 and part 2.

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: extend jobs timeouts (#11)

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: add new timeout to static build

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: update the badges in the README.md file

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: limit Static builds to ICX runners only

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

janekmi committed:
test: expand the sds.at_create CTL to also cover pmemobj_open() (fix)

- add missing log entry

Ref: daos-stack/pmdk#14

Signed-off-by: Jan Michalski <jan-marian.michalski@hpe.com>
Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: 2.1.2-rc1 release

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>
